app/models/miq_group.rb

```diff
@@ -18,17 +22,20 @@ class MiqGroup < ActiveRecord::Base
 
   delegate :self_service?, :limited_self_service?, :to => :miq_user_role, :allow_nil => true
 
-  validates_presence_of   :description, :guid
-  validates_uniqueness_of :description, :guid
+  validates :description, :guid, :presence => true, :uniqueness => true
+  validate :validate_default_tenant, :on => :update, :if => :tenant_id_changed?
 
   before_destroy do |g|
```

Comment:
Prefer a private method and using a symbol with before_destroy over passing a block

Not sure if still an issue, but in the past, passing a block messed up inheritance chaining of callbacks.  Also, it's cleaner if you ever have to add a second callback.